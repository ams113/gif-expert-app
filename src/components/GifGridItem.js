import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { title = '', url = '' } ) => {

    const titleValid = title.trim() || 'Without title';

    return (
        <article className="card animate__animated animate__fadeIn">
            <div className="card-content">
                <img src={ url } alt={ titleValid } />
                <p>{ titleValid }</p>
            </div>
        </article>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };