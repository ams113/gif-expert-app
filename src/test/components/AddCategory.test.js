import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

describe('Testing components: AddCategory', () => {
    
    const setCategories = jest.fn();

    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    
    beforeEach( () => {
        
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    });

    test('should renders correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('should change the value of the input', () => {

        const input = wrapper.find('input');
        input.simulate('change', { target: { value: 'foo'} });

        expect( wrapper.find('input').prop('value') ).toBe('foo');
    });

    test('should not post info with event submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        
        expect( setCategories ).not.toHaveBeenCalled();

    });

    test('should call setCategories and clean text input', () => {

        const value = 'foo';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
        expect( wrapper.find('input').prop('value') ).toBe('');
    });
});


    

