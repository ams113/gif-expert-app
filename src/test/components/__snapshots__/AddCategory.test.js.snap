// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components: AddCategory should renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
</form>
`;
